br/pkg/task/restore.go

```diff
@@ -1743,19 +1758,19 @@ func AdjustTablesToRestoreAndCreateTableTracker(
 	}
 
 	// first handle table renames to determine which tables we need
-	handleTableRenames(logBackupTableHistory, snapshotDBMap, cfg, tableMap, dbMap, piTRIdTracker)
-
-	// handle partition exchange after all tables are tracked
-	if err := handlePartitionExchanges(logBackupTableHistory, snapshotDBMap, cfg); err != nil {
-		return err
-	}
+	processLogBackupTableHistory(logBackupTableHistory, snapshotDBMap, snapshotTableMap,
+		partitionMap, cfg, tableMap, piTRIdTracker)
 
 	// track all snapshot tables that's going to restore in PiTR tracker
 	for tableID, table := range tableMap {
 		piTRIdTracker.TrackTableId(table.DB.ID, tableID)
 	}
 
-	log.Info("pitr table tracker", zap.String("map", piTRIdTracker.String()))
+	// handle partition exchange after all tables are tracked
+	if err := checkPartitionExchangesViolations(logBackupTableHistory, snapshotDBMap,
```

Comment:
`table : partition` means table has a partition
`t2`(start) -> `t1 : t2`(end)
if t2 is filter out, and. t1 is filter in.
so that partition `t2` in end will be tracked because its parent table `t1` is filtered.
finally the check will be passed.

Reply:
discussed offline, the original partition will be caught in the change but not the standalone table